ios-sdkTests/MapViewControllerTests.swift

```diff
@@ -338,4 +339,366 @@ class MapViewControllerTests: XCTestCase {
+    XCTAssertTrue(delegate.sceneLoaded)
+  }
+  
+  func testDidCompleteLoadingCallsLoadDelegate() {
+    let delegate = TestLoadDelegate()
+    controller.loadDelegate = delegate
+    controller.mapViewDidCompleteLoading(controller.tgViewController)
+    XCTAssertTrue(delegate.loadingComplete)
+  }
+  
+  func testDidSelectLabelCallsSelectDelegate() {
+    let delegate = TestMapSelectDelegate()
+    controller.selectDelegate = delegate
+    controller.mapView(tgViewController, didSelectLabel: TGLabelPickResult(), atScreenPosition: CGPoint())
+    XCTAssertTrue(delegate.labelPicked)
+  }
+  
+  func testDidSelectFeatureCallsSelectDelegate() {
+    let delegate = TestMapSelectDelegate()
+    controller.selectDelegate = delegate
+    controller.mapView(tgViewController, didSelectFeature: [AnyHashable](), atScreenPosition: CGPoint())
+    XCTAssertTrue(delegate.featurePicked)
+  }
+  
+  func testDidSelectMarkerCallsSelectDelegate() {
+    let delegate = TestMapSelectDelegate()
+    controller.selectDelegate = delegate
+    controller.mapView(tgViewController, didSelectMarker: TGMarkerPickResult(), atScreenPosition: CGPoint())
+    XCTAssertTrue(delegate.markerPicked)
+  }
+  
+  func testDidSelectLabelDoesNotCallSelectDelegate() {
+    let delegate = TestMapSelectDelegate()
+    controller.selectDelegate = delegate
+    controller.mapView(tgViewController, didSelectLabel: nil, atScreenPosition: CGPoint())
+    XCTAssertFalse(delegate.labelPicked)
+  }
+  
+  func testDidSelectFeatureDoesNotSelectDelegate() {
+    let delegate = TestMapSelectDelegate()
+    controller.selectDelegate = delegate
+    controller.mapView(tgViewController, didSelectFeature: nil, atScreenPosition: CGPoint())
+    XCTAssertFalse(delegate.featurePicked)
+  }
+  
+  func testDidSelectMarkerDoesNotSelectDelegate() {
+    let delegate = TestMapSelectDelegate()
+    controller.selectDelegate = delegate
+    controller.mapView(tgViewController, didSelectMarker: nil, atScreenPosition: CGPoint())
+    XCTAssertFalse(delegate.markerPicked)
+  }
+}
+
+class TestPanDelegate : NSObject, MapPanGestureDelegate {
+  
+  var didPanMap = false
+  
+  func mapController(_ view: MapViewController, didPanMap displacement: CGPoint) {
+    didPanMap = true
+  }
+}
+
+class TestPinchDelegate : NSObject, MapPinchGestureDelegate {
+  
+  var didPinchMap = false
+  
+  func mapController(_ view: MapViewController, didPinchMap displacement: CGPoint) {
+    didPinchMap = true
+  }
+}
+
+class TestRotateDelegate : NSObject, MapRotateGestureDelegate {
+  
+  var didRotateMap = false
+  
+  func mapController(_ view: MapViewController, didRotateMap displacement: CGPoint) {
+    didRotateMap = true
+  }
+}
+
+class TestShoveDelegate : NSObject, MapShoveGestureDelegate {
+  
+  var didShoveMap = false
+  
+  func mapController(_ view: MapViewController, didShoveMap displacement: CGPoint) {
+    didShoveMap = true
+  }
+}
+
+class AllEnabledGestureDelegate : NSObject, MapGestureDelegate {
```

Comment:
I think I do need to sublass off of `NSObject` because `MapGestureDelegate` implements `NSObjectProtocol` (and so does `NSObject`). I have `MapGestureDelegate` implementing `NSObjectProtocol` so that we can check if the object responds to optional selectors. Maybe we further breakdown `MapGestureDelegate` into 4 separate protocols (1 for all the `shouldRespondTo` calls, and then 3 for the `didRespondTo` calls. That way only the first one would require `respondsTo` checks)? Or is there another solution that we could use?